vendor/engines/secure_rooms/app/controllers/secure_rooms/card_readers_controller.rb

```diff
@@ -0,0 +1,81 @@
+module SecureRooms
+
+  class CardReadersController < ApplicationController
+
+    admin_tab :all
+
+    layout "two_column"
+
+    before_action :authenticate_user!
+    before_action :check_acting_as
+    before_action :authorize_card_reader
+    before_action :init_current_facility
+    before_action :init_product
+    before_action :load_card_reader, except: [:index]
+
+    def initialize
+      @active_tab = "secure_rooms"
+      super
+    end
+
+    def index
```

Comment:
I think it will work better for this project if we defer some of these questions to the validation phase, where they can be discussed separately and their effect on budget can be more easily tracked.